modules/travel_pay/app/services/travel_pay/appointments_service.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module TravelPay
+  class AppointmentsService
+    def get_appointments_by_date(veis_token, btsss_token, params = {})
+      faraday_response = client.get_all_appointments(veis_token, btsss_token, { 'excludeWithClaims' => true })
+      raw_appointments = faraday_response.body['data'].deep_dup
+
+      appointments = filter_by_date(params['appt_datetime'], raw_appointments)
+
+      {
+        data: appointments || []
+      }
+    end
+
+    private
+
+    def filter_by_date(date_string, appointments)
+      if date_string.present?
+        parsed_appt_date = Date.parse(date_string)
+
+        appointments.filter do |appointment|
+          appointment['appointmentDateTime'].nil? &&
+            parsed_appt_date == Date.parse(appointment['appointmentDateTime'])
+        end
+      end
+    rescue Date::Error => e
+      Rails.logger.debug(message: "#{e}. Unable to find appointments with date (given: #{date_string}).")
```

Comment:
I'm thinking in this case we should through an ArgumentError (or something) in order to keep the frontend as simple as possible. If we respond with an empty array, the frontend will have to check to see if there are entries. If we respond with a 400 - Bad Request, the frontend can jump straight to showing an error. Thoughts?